programming-languages.md

```diff
@@ -1,51 +1,82 @@
 ---
-title: Server-side programming languages
+title: Programming languages
 layout: recommendation
-expires: 2017-09-01
+expires: 2017-10-01
 ---
 
-All programming languages have their own strengths and weaknesses.
-This means that we want to use a range of programming languages in
-order to build services and supporting tools to meet users needs.
-This document does not apply to choosing 'off the shelf' software
-(open source or not).
+We want to use a range of programming languages at GDS because we think
+using the right tool for the job will give us the best chance of building
+services that best meet users’ needs. This document does not apply to
+choosing 'off the shelf' software (open source or not).
 
-In order to make it easier for developers to move around the organisation,
-to get more value out of developing shared components, to better hire and
-develop people, and to be great at operating applications, we want to focus
-on a small number of choices of programming languages for core
-'backend development' tasks.
+We want to focus on using a small number of programming languages for
+core software development tasks.
 
-These are (alphabetically): Java, Python, Ruby. These are successfully being
-used by teams at the moment, and we are confident in how to host and operate
-applications written in them. New development should normally be carried out
-in one of them.
+This should make it easier for developers to:
 
-There will be sensible reasons to go outside this core group. For example:
+- move around the organisation
+- develop shared components
+- improve their personal development
+- master how they operate applications
 
-- when we're extending an existing codebase/ecosystem
-- when scripting in a particular environment
-- when we're experimenting during an alpha (with an expectation that it's replaced by something from the core set for beta)
+## Java, Python and Ruby
```

Comment:
Can you make the headings clearer, so that you don't have to read the entire document to understand what's being said? (A quick scan would suggest our core languages are Java, Python, Ruby and also JavaScript, Go and Scala...)

So this would be `Our core languages are Java, Python and Ruby`. The heading for Scala would be `Do not use Scala for new projects`. Etc.